resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -391,7 +391,7 @@ private[spark] class Client(
   private[yarn] def copyFileToRemote(
       destDir: Path,
       srcPath: Path,
-      replication: Short,
+      replication: Option[Short],
```

Comment:
Do we have a test coverage for the new code path invoking `None`, @LuciferYang ?

Reply:
This is a good question. Let me add one